packages/react-native-renderer/src/ReactFabricHostConfig.js

```diff
@@ -343,7 +339,14 @@ export function getChildHostContext(
 }
 
 export function getPublicInstance(instance: Instance): * {
-  return instance.canonical;
+  if (instance.canonical) {
+    return instance.canonical;
+  }
+
+  // For compatibility with Paper
+  if (instance._nativeTag) {
+    return instance;
+  }
```

Comment:
This can return undefined but the places where getPublicInstance are called require an instance or null. I would have thought flow would error on this.

also, is `_nativeTag === 0` a special sentinel value? I assume this is actually true and why this logic makes sense but given I'm fairly unfamiliar with the native side I thought I'd point it out in case there was some oversight here

Reply:
> This can return undefined but the places where getPublicInstance are called require an instance or null. I would have thought flow would error on this.

I would've assumed so, yeah. I'll make this more robust.

> also, is _nativeTag === 0 a special sentinel value? I assume this is actually true and why this logic makes sense but given I'm fairly unfamiliar with the native side I thought I'd point it out in case there was some oversight here

Flow would've usually caught this too! I'll review why it didn't and fix it.

Thanks for the feedback!